### PR TITLE
test: add tests relating to job attachment failures and empty output

### PR DESCRIPTION
**Pull request by YutongLi291**
### What was the problem/requirement? (What/Why)
For when the customer is syncing their job attachments, either while syncing the inputs, or the outputs, we need to make sure that the sync works as expected.

In particular, the functionalities that:
* If an input file is not found, but the customer specified that all input paths should exist, we should fail the job submission, as this indicates that the customer likely forgot to save an input file 
* If there are no output files found in the output directory, the job should still pass.  This path should not have any regressions in the line
### What was the solution? (How)
* Add two tests that test these functionalities
* Also added a lot more typing to the tests file
### What is the impact of this change?
Better code coverage of the current features in worker-agent.
### How was this change tested?
```
# Linux
source .e2e_linux_infra.sh
hatch run linux-e2e-test
hatch run cross-os-e2e-test

# Windows
source .e2e_windows_infra.sh
hatch run windows-e2e-test
hatch run cross-os-e2e-test
```
### Was this change documented?
No
### Is this a breaking change?
No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: test/e2e/test_job_submissions.py [hunk 1]
```diff
@@ -1289,6 +1290,163 @@ def test_worker_uses_job_attachment_configuration(
...
+                            ],
+                        }
+                    )
+                )
+
+            config = configparser.ConfigParser()
+
+            set_setting("defaults.farm_id", deadline_resources.farm.id, config)
+            set_setting("defaults.queue_id", deadline_resources.queue_a.id, config)
+            with pytest.raises(Exception):
```
Comment (moorec-aws) [inline test/e2e/test_job_submissions.py]: Isn't this just testing the deadline cloud API? What are we testing for the worker here?

Comment (AlexTranAmz) [reply test/e2e/test_job_submissions.py]: same question

Comment (YutongLi291) [reply test/e2e/test_job_submissions.py]: There's no testing for the worker in this case. It tests the job attachments instead so I think I will remove this test. Thanks for calling this out!

**Comment by YutongLi291** [comment]
SonarCloud failing due to the typing I added.

**Review by moorec-aws — DISMISSED** [review]

**Review by AlexTranAmz — APPROVED** [review]

**Review by jairaws — APPROVED** [review]
